vendor/bat-native-ledger/src/bat/ledger/internal/bat_contribution.cc

```diff
@@ -1032,42 +1033,50 @@ void BatContribution::GetContributeWinners(
     const unsigned int ballots,
     const std::string& viewing_id,
     const braveledger_bat_helper::PublisherList& list) {
-  unsigned int total_votes = 0;
-  std::vector<unsigned int> votes;
-  braveledger_bat_helper::Winners res;
+  braveledger_bat_helper::Winners winners =
+      GetStatisticalVotingWinners(ballots, list);
+  VotePublishers(winners, viewing_id);
+}
 
-  for (auto &item : list) {
-    if (item.percent_ <= 0) {
+bool BatContribution::GetStatisticalVotingWinner(
+    double dart,
+    const braveledger_bat_helper::PublisherList& list,
+    braveledger_bat_helper::WINNERS_ST* winner) {
+  double upper = 0.0;
+  for (const auto& item : list) {
+    upper += item.weight_;
+    if (upper < dart)
       continue;
-    }
 
-    braveledger_bat_helper::WINNERS_ST winner;
-    winner.votes_ = static_cast<unsigned int>(std::lround(
-        static_cast<double>(item.percent_) *
-        static_cast<double>(ballots) / 100.0));
```

Comment:
It appears that:

- Before, GetContributeWinners used `percent_` to determine the number of votes cast for each publisher.
- After, its replacement GetStatisticalVotingWinners uses `weight_` to determine the number of votes cast for each publisher.

This may or may not be correct -- I don't know which one is intended -- but it appears to me that with the change, `percent_` is no longer used, which is suspicious.

Can we sort out what the meaning of each number is, and can you add comments to the definition of `struct PUBLISHER_ST` in bat_helper.h explaining which is which?

Reply:
`weight_` is just more accurate representation of `percent_`. Example from the db:

![image](https://user-images.githubusercontent.com/9574457/59215084-a7fb0500-8bb8-11e9-8fb6-fb118667e162.png)

So using `weight_` is correct as we get more accurate results

Reply:
OK, cool!  Can we add a comment explaining this somewhere?  (Doesn't have to be in this commit but wouldn't hurt.)

Reply:
Yes, will do!